handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -1790,40 +1797,34 @@ public void run() {
      *         marked as success by this method
      */
     private boolean setHandshakeSuccessIfStillHandshaking() {
-        if (!handshakePromise.isDone()) {
-            setHandshakeSuccess();
-            return true;
-        }
-        return false;
+        return setHandshakeSuccess();
     }
 
     /**
      * Notify all the handshake futures about the successfully handshake
      */
-    private void setHandshakeSuccess() {
-        boolean notified = handshakePromise.trySuccess(ctx.channel());
-        SSLSession session = engine.getSession();
-
-        // There seems to be a bug in the SSLEngineImpl that is part of the OpenJDK that results in returning
-        // HandshakeStatus.FINISHED multiple times which is not expected. This only happens in TLSv1.3 so lets
-        // ensure we only notify once in this case.
-        //
-        // This is safe as TLSv1.3 does not support renegotiation and so we should never see two handshake events.
-        if (notified || !SslUtils.PROTOCOL_TLS_V1_3.equals(session.getProtocol())) {
+    private boolean setHandshakeSuccess() {
```

Comment:
Would be good to also add a javadoc comment about the new return values.

Reply:
done